charts/gardener/controlplane/charts/runtime/templates/apiserver/deployment.yaml

```diff
@@ -491,6 +501,15 @@ spec:
               audience: {{ .Values.global.apiserver.admission.mutatingWebhook.token.audience }}
           {{- end }}
       {{- end }}
+      {{- if .Values.global.apiserver.workloadIdentity.token.signing.secretName }}
+      - name: gardener-apiserver-workload-identity
+        secret:
+          secretName: {{ .Values.global.apiserver.workloadIdentity.token.signing.secretName }}
+      {{- else if .Values.global.apiserver.workloadIdentity.token.signing.key  }}
+      - name: gardener-apiserver-workload-identity
+        secret:
+          secretName: gardener-apiserver-workload-identity
```

Comment:
Make the `if-else` only around `secretName` field?

Reply:
I was thinking the same for a moment, but then the volume would become required while it should be optional, at least until the GEP is fully implemented. It would look weird if gardener users have to provide configuration that is actually not yet usable.

Reply:
I was thinking about something like this:

```yaml
      {{- if .Values.global.apiserver.workloadIdentity.token.signing }}
      - name: gardener-apiserver-workload-identity
        secret:
      {{- if .Values.global.apiserver.workloadIdentity.token.signing.secretName }}
          secretName: {{ .Values.global.apiserver.workloadIdentity.token.signing.secretName }}
      {{- else if .Values.global.apiserver.workloadIdentity.token.signing.key  }}
          secretName: gardener-apiserver-workload-identity
      {{- end }}
      {{- end }}
```

It doesn't make a big difference, just a suggestion to prevent repetition.

Reply:
Done